wc3-hw1-TranNgocLam/cypress/e2e/registrationForm.cy.js

```diff
@@ -0,0 +1,308 @@
+    it.only('TCRegister#5', () => {
+      cy.get('@registration').then((users) => {
+        registerForm
+          .inputLastName(users[2].lastName)
+          .inputFirstName(users[2].firstName)
+          .checkGender(users[2].gender)
+          .inputUserNumber(users[2].userNumber)
+          .clickSubmit()
+
+        confirmForm
+          .isCorrectName(users[2].firstName, users[2].lastName)
+          .isCorrectGender(users[2].gender)
+          .isCorrectMobile(users[2].userNumber)
+          .clickClose()
+      });
+    });
+
+    /*  TCRegister#6: Verify that user cannot submit with invalid data:
+      - Email lacks of '@' and email domain.
+      - Phone number has less than 10 digits. */
+    it.only('TCRegister#6', () => {
+      cy.get('@registration').then((users) => {
+        registerForm
+          .inputLastName(users[3].lastName)
+          .inputFirstName(users[3].firstName)
+          .inputEmail(users[3].email)
+          .checkGender(users[3].gender)
+          .inputUserNumber(users[3].userNumber)
+          .clickSubmit()
+
+        registerForm
+          .checkFirstNameValid(true)
+          .checkLastNameValid(true)
+          .checkEmailValid(false)
+          .isGenderChecked(true)
+          .checkMobileValid(false)
+
+      });
+    });
+
+    /*  TCRegister#7: Verify that user cannot submit with invalid data:
+      - First Name includes only a space character.
+      - Last Name includes only a space character.
+      - Date of birth is selected by a day after today.
+      - Picture is selected with a file that is not an image format file. */
+    it.only('TCRegister#7', () => {
+      cy.get('@registration').then((users) => {
+        let tomorrow = new Date(new Date().getTime() + 24 * 60 * 60 * 1000);
+        let year = new Intl.DateTimeFormat('en', { year: 'numeric' }).format(tomorrow);
+        let month = new Intl.DateTimeFormat('en', { month: 'long' }).format(tomorrow);
+        let day = new Intl.DateTimeFormat('en', { day: '2-digit' }).format(tomorrow);
+        registerForm
+          .inputLastName(users[4].lastName)
+          .inputFirstName(users[4].firstName)
+          .inputEmail(users[4].email)
+          .checkGender(users[4].gender)
+          .inputUserNumber(users[4].userNumber)
+          .chooseDOB(day, month, year)
+          .uploadPicture(users[4].picture)
+          .clickSubmit()
+
+        confirmForm
+          .isCorrectName(users[4].firstName, users[4].lastName)
+          .isCorrectGender(users[4].gender)
+          .isCorrectEmail(users[4].email)
+          .isCorrectMobile(users[4].userNumber)
+          .isCorrectDOB(day, month, year)
+          .isCorrectPicture(users[4].picture)
+          .clickClose()
+
+      });
+    });
+
+    /*  TCRegister#8: Verify that user cannot submit with invalid data:
+      - Email lacks of email domain.
+      - Phone number includes characters. */
+    it.only('TCRegister#8', () => {
```

Comment:
I see that you implement many negative cases with the same steps, I recommend to re structure you test data so that you can make a loop, here some example:
```json
{
  "email": {
    "invalid": ["", ""]
  }
}
```
OR use this if you want to loop with many fields
```json
{
  "invalid": [
    {
      "email": "",
      "mobile": ""
    },
    {
      "email": "",
      "mobile": ""
    }
  ]
}
```